### ISSUE TITLE
parenthesis and indexing issue

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#699]
This is a simplified version of a user error:

```
code <- nimbleCode({
    y[1,1:3] <- omega[1,(1:(p*(p-1)/2))]
})

m = nimbleModel(code, constants = list(p=3))
Error: Error processing model: something is wrong with the index (1:3). Note that any variables in index expressions must be provided as constants.  NIMBLE does not yet allow indices that are model nodes.
```

Whereas this works fine:
```
code <- nimbleCode({
    y[1,1:3] <- omega[1,1:(p*(p-1)/2)]
})
```
The extra parentheses presumably trigger NIMBLE to see this as a functional operation rather than indexing with each index possibly the result of functions operating on constants. 

Just flagging this as something that arose; not clear there is anything we want to do, though there might be scope for a better error message here.

**Comment by perrydv** [comment]
This issue also causes a problem on the LHS:

```{r}
code <- nimbleCode({
    y[1,(1:3)] <- omega[1,1:(p*(p-1)/2)]
})

m = nimbleModel(code, constants = list(p=3))
```

**Comment by perrydv** [comment]
This was fixed by PR #709.
